core/src/env.rs

```diff
@@ -250,4 +251,10 @@ pub trait Ext {
 
     /// Get runtime cost weight.
     fn runtime_cost(&self, costs: RuntimeCosts) -> u64;
+
+    /// Pre-process memory access if need.
+    fn pre_process_memory_accesses(
```

Comment:
Consider move it into `IntoExtInfo` so you can move `AccessError` into backend

Reply:
Done